compose/service.py

```diff
@@ -936,20 +937,55 @@ def build_volume_binding(volume_spec):
     return volume_spec.internal, "{}:{}:{}".format(*volume_spec)
 
 
+def ensure_engine_compatibility(external_path, internal_path):
```

Comment:
Minor naming nit: this function name seems a lot more general then what it's doing. "ensure" to me means "validate", and "engine compatibility" is a much larger issue than just paths.

How about `normalize_host_path()`  (or something along those lines) ?